### ISSUE TITLE
Possibility of adding Playwright-bdd extension in to playwright.conf.ts

### ISSUE THREAD

**Issue by PavanMudigondaTR** [vitalets/playwright-bdd#4]
Currently we need to run two commands together "npx bddgen && npx playwright test".  Can we configure the extension in to playwright.conf.ts like we configure various reporter extensions ?  I understand there is no option to add third party extensions in Playwright Test currently other than reporters section. Can you add "playwright-bdd" in to reporter section similar to this [Playwright Azure DevOps TestPlans Extension](https://github.com/alexneo2003/playwright-azure-reporter). This will make users run one command directly.

One problem to think of here is how will VS Code Extension work ? Can it identify the tests  since tests are not auto generated at this point ? Can Playwright team make change to "VS Code Playwright extension" to  run "npx bddgen" command based on "playwright.conf.ts" and detect the tests ?

**Comment by vitalets** [comment]
I was thinking about embedding playwright-bdd into `playwright.conf.ts` to keep things simple. 

Then I found this https://github.com/microsoft/playwright/issues/12857#issuecomment and this https://github.com/microsoft/playwright/issues/14705#issuecomment-1150017505 from playwright team saying that the recommended way to pre-generate tests is to do it outside of playwright process. The main reason is exactly as you mentioned - extensions can work only with statically generated test files.

I they will add  'generator' hook it would be definitely used here.

**Comment by vitalets** [comment]
I've got some new thoughts on this. 
Although we still need to run `npx bddgen` as a separate command, it's good idea to embed cucumber config into playwright config. For example:

```ts
// playwright.config.ts
import { defineConfig } from '@playwright/test';
import ( configureBDD } from 'playwright-bdd';

const testDir = configureBDD({
  paths: ['test/features/*.feature'],
  require: ['test/steps*.ts'],
  requireModule: ['ts-node/register'],
});

export default defineConfig({
  testDir,
  // ...
});
```

The benefits are:

1. one config file instead of two
2. possibility to change `bddgen` outputDir in single place
3. eventually we can support playwright projects for `bddgen` to generate different tests for different projects:
   ```ts
   // playwright.config.ts
   export default defineConfig({
      projects: [
       { 
           name: 'e2e', 
           testDir: configureBDD({
             outputDir: './e2e/.features-gen',
             paths: ['./e2e/features'],
             require: ['./e2e/steps'],
          })
       },
       { 
           name: 'acceptance', 
           testDir: configureBDD({
             outputDir: './acceptance/.features-gen',
             paths: ['./acceptance/features'],
             require: ['./acceptance/steps'],
          })
       },
     ],
      // ...
   });
   ```

Under the hood `bddgen` will read playwright config but do not run playwright itself. On real playwright run `configureBDD` should just return `testDir` without any test generation.

**Comment by vitalets** [comment]
Implemented single configuration file in v3.0.1.
Also added [note](https://github.com/vitalets/playwright-bdd#faq) in README about having separate command for tests generation.
